app/models/manageiq/providers/microsoft/infra_manager/ps_scripts/get_inventory.ps1

```diff
@@ -1,103 +1,45 @@
-}
-
-$r = @{}
-$v = Get-SCVirtualMachine -VMMServer "localhost"
-
-$r["vms"] = get_vms($v)
-
-$i = Get-SCVMTemplate -VMMServer "localhost"
-$r["images"] = get_images($i)
-
-$h = Get-SCVMHost -VMMServer "localhost"
-$r["hosts"] = get_host_inventory($h)
-$r["datastores"] = $diskvols
-
-$c = Get-SCVMHostCluster -VMMServer "localhost"
-$r["clusters"] = get_clusters($c)
-
-$e = Get-SCVMMServer -ComputerName "localhost"
-$r["ems"] = $e
-
-$inFile = [System.IO.Path]::GetTempFileName()
-$outFile = $inFile + '.gz'
-
-Export-CLIXML -Input $r -Path $inFile -Encoding UTF8
-
-$in = New-Object System.IO.FileStream $inFile, ([IO.FileMode]::Open), ([IO.FileAccess]::Read), ([IO.FileShare]::Read)
-$buf = New-Object byte[]($in.Length)
-$x = $in.Read($buf, 0, $in.Length)
-$in.Dispose()
-
-$out = New-Object System.IO.FileStream $outFile, ([IO.FileMode]::Create), ([IO.FileAccess]::Write), ([IO.FileShare]::None)
-
-$gStream = New-Object System.IO.Compression.GzipStream $out, ([IO.Compression.CompressionMode]::Compress)
-$x = $gStream.Write($buf, 0, $buf.Length)
-$gStream.Dispose()
-$out.Dispose()
-
-[System.convert]::ToBase64String([System.IO.File]::ReadAllBytes($outFile))
-
-Remove-Item -Force $inFile
-Remove-Item -Force $outFile
+$hash = @{}
+
+$ems = Get-SCVMMServer -ComputerName localhost |
+  Select @{name='Guid';expression={$_.ManagedComputer.ID -As [string]}},
+    @{name='Version';expression={$_.ServerInterfaceVersion -As [string]}}
+
+$vms = Get-SCVirtualMachine -VMMServer localhost -All |
+  Select -Property BackupEnabled,BiosGuid,ComputerName,CPUCount,CPUType,DataExchangeEnabled,
+    HeartbeatEnabled,HostName,ID,LastRestoredCheckpointID,Memory,Name,OperatingSystem,
+    OperatingSystemShutdownEnabled,ServerConnection,TimeSynchronizationEnabled,
+    VirtualDVDDrives,VirtualHardDisks,VirtualNetworkAdapters,VMCheckpoints,VMCPath,
+    @{name='StatusString';expression={$_.Status -As [string]}},
+    @{name='VirtualMachineStateString';expression={$_.VirtualMachineState -As [string]}},
+    @{name='VirtualDVDDrivesISO';expression={ $_.VirtualDVDDrives.ISO }}
+
+$hosts = Get-SCVMHost -VMMServer localhost |
+  Select -Property CommunicationStateString,CoresPerCPU,DiskVolumes,DVDDriveList,
+    HyperVStateString,ID,LogicalProcessorCount,Name,OperatingSystem,PhysicalCPUCount,
+    ProcessorFamily,ProcessorManufacturer,ProcessorModel,ProcessorSpeed,TotalMemory,
+    @{name='HyperVVersionString';expression={$_.HyperVVersion -As [string]}},
+    @{name='OperatingSystemVersionString';expression={$_.OperatingSystemVersion -As [string]}},
+    @{name='VirtualizationPlatformString';expression={$_.VirtualizationPlatform -As [string]}}
+
+$vnets = Get-SCVirtualNetwork -VMMServer localhost |
+  Select -Property ID,Name,LogicalNetworks,VMHostNetworkAdapters,
+    @{name='VMHostName';expression={$_.VMHost.Name -As [string]}}
+
+$images = Get-SCVMTemplate -VMMServer localhost -All |
+  Select -Property CPUCount,Memory,Name,ID,VirtualHardDisks,VirtualDVDDrives,

```

Comment:
The -All forces it to get subordinate objects so we can get at the properties we need.